### PR TITLE
Feature/bitmap

### PR DESCRIPTION
**Pull request by Shariefie**
Check my stuff please and review thoroughly asap.

### CODE REVIEW

File: Core/Src/API.c [hunk 1]
```diff
@@ -1 +1,80 @@
...
+#include "API.h"
+
+/**
+ *@brief Draw bitmap function, this function draws the bitmap smiley, angry and the four arrows on the screen.
+ *			The user can choose between smiley=0, angry=1, left=2, up=3, right=4, right=5.
+ *
+ *@param nr is the bitmap number to draw.
+ *@param x_1up and y_1up is the position of the bitmap
+ * */
+void draw_bitmap(int nr, int x_1up, int y_1up)
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: as discussed change function name to drawBitmap

File: Core/Src/API.c [hunk 2]
```diff
@@ -1 +1,80 @@
...
+/**@brief Draw text function, this function prints the text on the screen.
+ * 			The user can choose between Arial= 0 en Consolas=1.
+ *@param x, y is the position of the starting point of the first letter.
+ *@param kleur is the fonts colour.
+ *@param tekst[] is an array with text.
+ *@param fontname can be chosen from fonts of Arial and Consolas.
+ *@param fontgrootte is the size of the font. The size can be chosen from 8 or 32. 0 = 8 and 1 = 32.
+ *@param frontstyle is the style of fonts, this can be chosen of Italic, Bold.
+ * */
+void draw_text(int x, int y, char kleur, char tekst[], char fontname[], char fontgrootte, char fontstyle)
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: same as draw bitmap

File: Core/Src/API.c [hunk 3]
```diff
@@ -1 +1,80 @@
...
+	int i, j, k, l;
+	int x_old = 0;
+	const char *letter;		//Pointer to the array of letter
+	char width;				//The width of every letter can be different.
+
+	const char **fonts[] = {Arial_8_addr, Arial_32_addr, Arial_Italic_8_addr, Arial_Italic_32_addr, Arial_Bold_8_addr, Arial_Bold_32_addr,
+			Consolas_8_addr, Consolas_32_addr, Consolas_Italic_8_addr, Consolas_Italic_32_addr, Consolas_Bold_8_addr, Consolas_Bold_32_addr};
+
+	char font_name[3][20] = {"Arial", "Consolas"};	//Names of fonts in array.
+	const char **font;
```
Comment (bestehustudentEU) [inline Core/Src/API.c:323]: two pointers named font and fonts are confusing

File: Core/Inc/API.h [hunk 4]
```diff
@@ -1 +1,27 @@
...
+#include "fonts/Consolas_Italic_32.h"
+#include "fonts/Consolas_Italic_8.h"
+#include "fonts/Consolas_Bold_32.h"
+#include "fonts/Consolas_Bold_8.h"
+#include "stm32_ub_vga_screen.h"
+
+#define BEGGINING_OF_BITMAP 3
+#define BITMAP_WIDTH_HIGH	2
+#define BITMAP_WIDTH_LOW	1
+
```
Comment (bestehustudentEU) [inline Core/Inc/API.h:152]: no comments?

Comment (bestehustudentEU) [reply Core/Inc/API.h:152]: i meant debug :p

File: Core/Src/API.c [hunk 5]
```diff
@@ -1 +1,80 @@
...
+  * @brief          : Library for Graphics.
+  ******************************************************************************
+  */
 
+#include "API.h"
+
+/**
+ *@brief Draw bitmap function, this function draws the bitmap smiley, angry and the four arrows on the screen.
+ *			The user can choose between smiley=0, angry=1, left=2, up=3, right=4, right=5.
+ *
```
Comment (bestehustudentEU) [inline Core/Src/API.c:267]: no details?(doxygen)

File: Core/Src/API.c [hunk 6]
```diff
@@ -1 +1,80 @@
...
+ *@param nr is the bitmap number to draw.
+ *@param x_1up and y_1up is the position of the bitmap
+ * */
+void draw_bitmap(int nr, int x_1up, int y_1up)
+{
+	int i, j;
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: initialize i and j

File: Core/Src/API.c [hunk 7]
```diff
@@ -1 +1,80 @@
...
+ *@param x, y is the position of the starting point of the first letter.
+ *@param kleur is the fonts colour.
+ *@param tekst[] is an array with text.
+ *@param fontname can be chosen from fonts of Arial and Consolas.
+ *@param fontgrootte is the size of the font. The size can be chosen from 8 or 32. 0 = 8 and 1 = 32.
+ *@param frontstyle is the style of fonts, this can be chosen of Italic, Bold.
+ * */
+void draw_text(int x, int y, char kleur, char tekst[], char fontname[], char fontgrootte, char fontstyle)
+{
+	int i, j, k, l;
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: initialize!

File: Core/Src/API.c [hunk 8]
```diff
@@ -1 +1,80 @@
...
+		if(!strcmp(font_name[i], fontname))			//If (font_name == font_name) output is 0.
+		{
+			font = fonts[i*FONT_TYPES+fontgrootte+fontstyle*FONT_SIZE];
+			width_array = *font;
+			break;
+		}
+	}
+	for(i = 0; i < strlen(tekst); i++)
+	{
+		letter = font[tekst[i]-31];				//The first character spacebar.
```
Comment (bestehustudentEU) [inline Core/Src/API.c]: could you use #define?

**Review by bestehustudentEU — CHANGES_REQUESTED** [review]
there not that many comments but do not know if it neccessary to put more.

File: Core/Src/API.c [hunk 9]
```diff
@@ -1 +1,80 @@
...
+	const char *letter;		//Pointer to the array of letter
+	char width;				//The width of every letter can be different.
+
+	const char **fonts[] = {Arial_8_addr, Arial_32_addr, Arial_Italic_8_addr, Arial_Italic_32_addr, Arial_Bold_8_addr, Arial_Bold_32_addr,
+			Consolas_8_addr, Consolas_32_addr, Consolas_Italic_8_addr, Consolas_Italic_32_addr, Consolas_Bold_8_addr, Consolas_Bold_32_addr};
+
+	char font_name[3][20] = {"Arial", "Consolas"};	//Names of fonts in array.
+	const char **font;
+	const char *width_array;
+	for(i = 0; i<2; i++)
```
Comment (Tjerk258) [inline Core/Src/API.c]: Define for number of fonts

File: Core/Src/API.c [hunk 10]
```diff
@@ -1 +1,80 @@
...
+				for(l=0; l < BYTE_SIZE; l++)
+				{
+					if ((letter[j*width+k] >> l) & 0x01)	//Compare the LSB with 0x01 (AND) if so, then print then print pixel (font) on the screen.
+					{
+						UB_VGA_SetPixel(x+x_old + k, y+j*BYTE_SIZE+l, kleur);
+					}
+				}
+			}
+		}
+		x_old += k + 2;	// draw 2 pixels between the letters.
```
Comment (Tjerk258) [inline Core/Src/API.c]: Define voor de 2

**Review by Tjerk258 — CHANGES_REQUESTED** [review]

File: Core/Src/API.c (hunk 2 above)
Comment (Shariefie) [reply Core/Src/API.c]: issue solved.

File: Core/Src/API.c (hunk 3 above)
Comment (Shariefie) [reply Core/Src/API.c:323]: I find it clear enough.

File: Core/Src/API.c (hunk 5 above)
Comment (Shariefie) [reply Core/Src/API.c:267]: Issue solved.

File: Core/Src/API.c (hunk 6 above)
Comment (Shariefie) [reply Core/Src/API.c]: Issue solved.

File: Core/Src/API.c (hunk 7 above)
Comment (Shariefie) [reply Core/Src/API.c]: Issue solved.

File: Core/Src/API.c (hunk 8 above)
Comment (Shariefie) [reply Core/Src/API.c]: Solved.

File: Core/Src/API.c (hunk 9 above)
Comment (Shariefie) [reply Core/Src/API.c]: Solved.

File: Core/Src/API.c (hunk 10 above)
Comment (Shariefie) [reply Core/Src/API.c]: Solved.

**Pull request merged**